src/call/mod.rs

```diff
@@ -31,8 +31,58 @@ use crate::codec::{DeserializeFn, Marshaller, SerializeFn};
 use crate::error::{Error, Result};
 use crate::task::{self, BatchFuture, BatchType, CallTag, SpinLock};
 
-pub use crate::grpc_sys::grpc_status_code as RpcStatusCode;
-pub type RpcStatusCodeType = i32;
+/// An gRPC status code structure.
+/// This type contains constants for all gRPC status codes.
+#[derive(PartialEq, Clone, Copy, Debug)]
+pub struct RpcStatusCode(i32);
+
+impl From<i32> for RpcStatusCode {
```

Comment:
Do we need this `From` impl? If we don't have it, then we can guarantee that `RpcStatusCode`  only has valid values.

Reply:
I don't think there is such a valid range for code, users are free to use whatever code they want. https://github.com/grpc/grpc/blob/master/doc/statuscodes.md

Reply:
convert to `new()` and assert in this function